src/components/settings/SettingsForm.tsx

```diff
@@ -0,0 +1,316 @@
+import classNames from 'classnames';
+import { CopyIcon, EyeIcon, EyeOffIcon } from 'lucide-react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useNotification } from '../../hooks/useNotification';
+import { useSessionStorage } from '../../hooks/useSessionStorage';
+import { decode } from '../../node/encoding';
+import { setSettingsApiKey } from '../../node/settings';
+import { getSettings } from '../../node/settings';
+import type { Pin } from '../../types';
+import { handleLinkClick } from '../../utils';
+import Button from '../common/Button';
+import Notification from '../common/Notification';
+import Description from '../typography/Description';
+import Label from '../typography/Label';
+import PageHeading from '../typography/PageHeading';
+
+export default function SettingsForm() {
```

Comment:
* If api key is set, you could not display form, but have only an option to change API key
   * So display information only API key set + is using PIN
   * and option to edit
* Once I want to edit the key display the form but without any existing api key
  * now for example you even show the encrypted API key, which makes no sense at all

![image](https://github.com/user-attachments/assets/8e6d3d6f-2fdf-4501-b8bc-4a68bcd30cad)
 


![image](https://github.com/user-attachments/assets/d6c94eb7-9fb5-44c1-b7c0-27e5760a807a)